### ISSUE TITLE
ASE won't apply an update

### ISSUE THREAD

**Issue by InteXX** [microsoft/AzureStorageExplorer#4903]
### Preflight Checklist

- [ ] I have installed the [latest version of Storage Explorer](https://github.com/Microsoft/AzureStorageExplorer/releases/latest).
- [X] I have checked existing resources, including the [troubleshooting guide](https://docs.microsoft.com/azure/storage/common/storage-explorer-troubleshooting) and the [release notes](https://github.com/microsoft/AzureStorageExplorer/releases).
- [X] I have searched for [similar issues](https://github.com/Microsoft/AzureStorageExplorer/issues).

### Storage Explorer Version

1.20.0

### Regression From

1.19.1

### Architecture

i86

### Storage Explorer Build Number

20210625.5

### Platform

Windows

### OS Version

Windows 10

### Bug Description

I'm having trouble getting ASE to apply an update (v1.20.0 -> v1.20.1).

When the download is complete I'm presented with the standard choices to proceed:

![image](https://user-images.githubusercontent.com/10966749/132605509-fb067620-b8ea-4488-91ed-95953bbca98b.png)

However, when I choose **Install Now** the application exits without applying the update. When I restart the application, it again downloads the same update and I'm caught in a loop.

### Steps to Reproduce

1. Launch Storage Explorer
2. Wait for the update download to complete
3. Click **Install Now** to attempt to apply the update
4. Restart the application after it exits
5. Note that the application wasn't updated and the update is downloaded again

### Actual Experience

See the **Steps to Reproduce** step above

### Expected Experience

I was expecting the application to apply the update.

### Additional Context

This might be related to the fact that earlier I mistakenly clicked the **Opt in to Insider Builds** menu item. That download managed to proceed a ways before I turned off the option and rolled back to the production release.

Is there a way to 'reset' the updating engine and start over from scratch?

**Comment by jinglouMSFT** [comment]
Can you try this:
1. Close Storage Explorer
2. Delete %AppData%/StorageExplorer
3. Restart Storage Explorer.

Notice that this will reset everything for your Storage Explorer.  
If you have ever imported any SSL certificates, back up the contents of the certs directory before the delete. Later, you can use the backup to reimport your SSL certificates.

**Comment by InteXX** [comment]
@jinglouMSFT

Thank you, but in the meantime I tried choosing the **Install on Close** option. That worked.

I'm now running v1.21.0, so we'll have to wait until the next update is released before we can test again.

**Comment by jinglouMSFT** [comment]
Install Now technically just close the Storage Explorer, then run the setup.  Did you see Storage Explorer being closed then setup being run?  
Glad it's up running for you now.  v1.21.0 was just released yesterday. You are on the latest now.

**Comment by InteXX** [comment]
I think what was happening was that **Install Now** was indeed closing ASE but not running the installer for some reason. I never saw Setup's UAC prompt.

I'll close the issue for now, but if it happens again I'll reopen it. Will that be OK with you?

**Comment by jinglouMSFT** [comment]
Sounds good.  Please feel free to either open a new issue or reactivate this one should you run into the same issue again.  Thanks.

**Comment by InteXX** [comment]
Will do, thanks.
